plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSplitManager.java

```diff
@@ -40,11 +39,7 @@
     public MongoSplitManager(MongoClientConfig config, MongoSession session)
     {
         if (config.getSeeds().isEmpty()) {
-            List<ServerDescription> descriptions = session.getClient().getClusterDescription().getServerDescriptions();
-
-            this.addresses = descriptions.stream()
-                    .map(s -> fromParts(s.getAddress().getHost(), s.getAddress().getPort()))
-                    .collect(toList());
+            this.addresses = session.getAddresses();
```

Comment:
I assume we can use `session.getAddresses()` regardless of seeds/connection-uri. Please let me know if it didn't work.